modules/glimpse/chunk/main.nf

```diff
@@ -19,20 +19,16 @@ process GLIMPSE_CHUNK {
     task.ext.when == null || task.ext.when
 
     script:
-    def prefix          = task.ext.prefix ?: "${meta.id}"
-    def thread_command  =                    "--thread $task.cpus"
-
-    def input_command   = reference_panel ? "--input $reference_panel" : ""
-    def region_command  = region          ? "--region $region"         : ""
-
+    def prefix  = task.ext.prefix ?: "${meta.id}"
+    def args    = task.ext.args   ?: ""
     def VERSION = '1.1.1' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
 
     """
     GLIMPSE_chunk \\
         $args \\
-        $input_command \\
-        $region_command \\
-        $thread_command \\
+        --input $input \\
+        --region $region \\
```

Comment:
Is region a mandatory option? If not, you could keep the previous code for the region

Reply:
It is a mandatory option.
The thing is that function ask for a specific region in the genome (generally a chromosome: eg 'chr21') and will cut this whole region in small part ready to use for the next function of the software.

Reply:
allright, this is perfect then!

Reply:
I don't quite understand where I and when I should add the non-mandatory options.
Do I need to explicitly put them if there is a need for a control ?

Reply:
If a option is not mandatory you need to add a check to see if the options has a value (Nextflow does not natively support optional inputs, so you need to enter `[]` for the inputs that should be empty). The checks are exactly like you did with `region_command` for example. You first check if the `region` is specified, if it is specified you tell it to add this parameter with the input to the command, otherwise you just let it add an empty string that does nothing.

Reply:
And what is exactly the difference with `args` ?
Because there is plenty of non mandatory arguments that I could add ([here](https://odelaneau.github.io/GLIMPSE/commands.html)).
Do I need to specify them ?

Reply:
`args` is mainly used to add arguments that do not need an input file. Although you can do some magic stuff with it, but you won't have to do that very often.